### ISSUE TITLE
manage '*' version as latest 

### ISSUE THREAD

**Issue by JR-Utily** [pnpm/pnpm#865]
Hello, 

I have a dependency that itself have a dependency declared with version = "*"
I can not modify it myself, this is just what I get

With npm, the resolve is done as "get latest", but with pnpm, it failed and tells "No compatible version found" then list the versionsin the registry.

This is easily reproducible with 

```
"devDependencies": {
    "local-dependencies": "^1.0.0-beta3"
} 

```

**Comment by zkochan** [comment]
This is an interesting case. [semver calculator](https://semver.npmjs.com/) says that `*` does not match any of the available versions of `inquirer-path`.

![Imgur](http://i.imgur.com/Ukhts9h.png)

`npm@5.3.0` installs `inquirer-path` and Yarn suggests selecting the needed version.

```
? Please choose a version from this list: 
  1.0.0-beta3 
❯ 1.0.0-beta2 
  1.0.0-alpha5 
  1.0.0-alpha4 
  1.0.0-alpha3 
  1.0.0-alpha2 
  1.0.0-alpha1 
```

I guess npm treats `*` as a latest tag and that is the easiest solution, so lets' do the same in pnpm

**Comment by zkochan** [comment]
Fix published in [v1.10.0](https://github.com/pnpm/pnpm/releases/tag/v1.10.0)
